components/script/dom/bindings/codegen/CodegenRust.py

```diff
@@ -5247,12 +5247,18 @@ def __init__(self, descriptor, constructor=None):
             assert constructor
         CGAbstractExternMethod.__init__(self, descriptor, name, 'bool', args)
         self.constructor = constructor
+        self.exposureSet = descriptor.interface.exposureSet
 
     def definition_body(self):
-        preamble = CGGeneric("""\
-let global = GlobalScope::from_object(JS_CALLEE(cx, vp).to_object());
-let args = CallArgs::from_vp(vp, argc);
-""")
+        preamble = """let global = GlobalScope::from_object(JS_CALLEE(cx, vp).to_object());\n"""
+        if len(self.exposureSet) == 1:
+            globalName = list(self.exposureSet)[0]
+            downcast = """let global = global.downcast::<dom::{}::{}>().unwrap();\n"""
+            downcast = downcast.format(globalName.lower(),
+                                       globalName[0].upper() + globalName[1:])
```

Comment:
This is for the case when a DOM type is exposed to exactly one global. In that case, we downcast `global` from `GlobalScope` to that particular struct.

Thank you for pointing out `MakeNativeName`! I have fixed it.